### PR TITLE
Reformat LBANN code base

### PR DESCRIPTION
**Pull request by bvanessen**
Reformatted the code base using the following command:
`clang-format -i $(git ls-files src include model_zoo/lbann*.cpp | grep -E "(pp|cu)$")`

### CODE REVIEW

**Comment by bvanessen** [comment]
@benson31 @tbennun @ndryden I have compiled this and I am currently working to test this.  However, it should be ready to review.

File: include/lbann/callbacks/checkpoint.hpp [hunk 1]
```diff
@@ -62,7 +63,8 @@ class checkpoint : public callback_base {
    *  @param checkpoint_epochs interval to checkpoint
    *  @param checkpoint_steps interval to checkpoint
    *  @param checkpoint_secs interval to checkpoint
-   *  @param per_rank_dir The directory into which to dump distributed checkpoints
+   *  @param per_rank_dir The directory into which to dump distributed
+   * checkpoints
```
Comment (tbennun) [inline include/lbann/callbacks/checkpoint.hpp:67]: Can we change the style to be aligned with "The" in the line above?

**Review by tbennun — APPROVED** [review]
Mainly a style change, so not much to say. I have two comments:
1. About the documentation style, see inline comment
2. There are going to be a few leftover copyrights (.clang-format for example) because their end year is not 2022. There aren't too many of those.

**Review by benson31 — APPROVED** [review]
LGTM

**Review by ndryden — APPROVED** [review]
Looks good to me.

**Comment by bvanessen** [comment]
There are no new CI failures.

**Pull request merged**